mithril-test-lab/mithril-end-to-end/src/end_to_end_spec.rs

```diff
@@ -180,6 +180,30 @@ impl<'a> Spec<'a> {
             assertions::assert_client_can_verify_snapshot(&mut client, &digest).await?;
         }
 
+        // Verify that Cardano database snapshot artifacts are produced and signed correctly
+        {
+            let merkle_root =
+                assertions::assert_node_producing_cardano_database_snapshot(&aggregator_endpoint)
+                    .await?;
+            let certificate_hash = assertions::assert_signer_is_signing_cardano_database_snapshot(
+                &aggregator_endpoint,
+                &merkle_root,
+                expected_epoch_min,
+            )
+            .await?;
+
+            assertions::assert_is_creating_certificate_with_enough_signers(
+                &aggregator_endpoint,
+                &certificate_hash,
+                self.infrastructure.signers().len(),
+            )
+            .await?;
+
+            // TODO: uncomment when the client can verify Cardano database snapshots
+            //let mut client = self.infrastructure.build_client()?;
+            //assertions::assert_client_can_verify_cardano_database_snapshot(&mut client, &digest).await?;
```

Comment:
IMHO, I don’t think this commented code is really necessary.

Reply:
Let's keep it as it will be easier to do the verification in the e2e once it is implemented in the client